tokio/src/sync/mpsc/bounded.rs

```diff
@@ -1036,6 +1036,37 @@ impl<T> Sender<T> {
             chan: self.chan.downgrade(),
         }
     }
+
+    /// Returns the max buffer capacity of the channel.
+    ///
+    /// The max capacity is the buffer capacity initially specified when calling
+    /// [`channel`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::sync::mpsc;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (tx, rx) = mpsc::channel::<()>(5);
+    ///      
+    ///     // both max capacity and capacity are the same at first
+    ///     assert_eq!(tx.max_capacity(), 5);
+    ///     assert_eq!(tx.capacity(), 5);
+    ///
+    ///     // Making a reservation doesn't change the max capacity.
+    ///     let permit = tx.reserve().await.unwrap();
+    ///     assert_eq!(tx.max_capacity(), 5);
+    ///     // but drops the capacity by one
+    ///     assert_eq!(tx.capacity(), 4);
```

Comment:
we may also want to show an example using `send` where the capacity is held by a message actually in the channel? that's maybe a slightly more common usage pattern than `reserve`. but, adding that isn't a big deal IMO, this example also gets the idea across.

Reply:
I've used `send` in tests/sync_mpsc.rs